### ISSUE TITLE
Bug while parsing C++ code with operators declarations

### ISSUE THREAD

**Issue by AlexNisnevich** [tree-sitter/tree-sitter-cpp#125]
_(originally posted at https://github.com/tree-sitter/tree-sitter/issues/1338)_

I've been playing around with the [tree-sitter playground](https://tree-sitter.github.io/tree-sitter/playground) and have been running into issues parsing some C++ files. I've narrowed the issue down to this minimum viable example:
```cpp
struct A {
	bool operator ==(A other) {
		return true;
	}
};

class B {
	bool operator ==(B other);
};
```

When pasting this valid C++ code into the playground, I get the following tree:
```
translation_unit [0, 0] - [10, 0]
  ERROR [0, 0] - [8, 2]
    type_identifier [0, 7] - [0, 8]
    field_declaration [1, 1] - [4, 2]
      type: primitive_type [1, 1] - [1, 5]
      declarator: field_identifier [1, 6] - [1, 14]
      ERROR [1, 15] - [1, 16]
      default_value: compound_literal_expression [1, 17] - [3, 2]
        ERROR [1, 18] - [1, 19]
          identifier [1, 18] - [1, 19]
        type: type_descriptor [1, 20] - [1, 25]
          type: type_identifier [1, 20] - [1, 25]
        value: initializer_list [1, 27] - [3, 2]
          identifier [2, 2] - [2, 8]
          ERROR [2, 9] - [2, 14]
            true [2, 9] - [2, 13]
      ERROR [4, 0] - [4, 1]
    field_declaration [6, 0] - [8, 2]
      type: class_specifier [6, 0] - [8, 1]
        name: type_identifier [6, 6] - [6, 7]
        body: field_declaration_list [6, 8] - [8, 1]
          field_declaration [7, 1] - [7, 27]
            type: primitive_type [7, 1] - [7, 5]
            declarator: field_identifier [7, 6] - [7, 14]
            ERROR [7, 15] - [7, 16]
            default_value: parenthesized_expression [7, 17] - [7, 26]
              ERROR [7, 18] - [7, 19]
                identifier [7, 18] - [7, 19]
              identifier [7, 20] - [7, 25]
```

Notably, both the `struct_specifier` and `class_specifier` are missing, and the top-level node (under `translation_unit`) is an `ERROR` node.

Interestingly, I can remove either the struct or the class from this example and the parse works fine, but having them both together somehow breaks the parser. 

It appears that the issue has something to do with the `==` operator declarations, because replacing the operator declarations with regular method declarations makes the parse work.

**Comment by theHamsta** [comment]
Did you test whether this would be fixed by https://github.com/tree-sitter/tree-sitter-cpp/pull/72

**Comment by AlexNisnevich** [comment]
This bug does indeed go away with #72. 👍 Do we have a sense of when #72 can be merged?

(Removing spaces next to the operator works as well, but is not really a viably solution for my use case.)

**Comment by theHamsta** [comment]
@calixteman do you have any plans to move the open PRs forward?

**Comment by theHamsta** [comment]
Fixed by #72 :tada:
